sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ModelTests.cs

```diff
@@ -0,0 +1,191 @@
+                {
+                    if (strategy.GetType().Name.StartsWith("ModelJsonConverterStrategy"))
+                    {
+                        //we never get to the interface implementation because JsonSerializer errors before that
+                        Assert.Throws<JsonException>(() => { T model = (T)strategy.Read(serviceResponse, ModelInstance, options); });
+                        result = true;
+                    }
+                    else
+                    {
+                        Assert.Throws<InvalidOperationException>(() => { T model = (T)strategy.Read(serviceResponse, ModelInstance, options); });
+                        result = true;
+                    }
+                }
+
+                if (strategy.IsExplicitJsonWrite)
+                {
+                    Assert.Throws<InvalidOperationException>(() => { var data = strategy.Write(ModelInstance, options); });
+                    result = true;
+                }
+            }
+            else if (ModelInstance is not IJsonModel<T> && format == "J")
+            {
+                Assert.Throws<FormatException>(() => { T model = (T)strategy.Read(serviceResponse, ModelInstance, options); });
+                Assert.Throws<FormatException>(() => { var data = strategy.Write(ModelInstance, options); });
+                result = true;
+            }
+            return result;
+        }
+
+        internal static Dictionary<string, BinaryData> GetRawData(object model)
+        {
+            Type modelType = model.GetType();
+            while (modelType.BaseType != typeof(object) && modelType.BaseType != typeof(ValueType))
+            {
+                modelType = modelType.BaseType!;
+            }
+            var propertyInfo = modelType.GetField("_rawData", BindingFlags.Instance | BindingFlags.NonPublic);
+            return propertyInfo?.GetValue(model) as Dictionary<string, BinaryData> ?? throw new InvalidOperationException($"unable to get raw data from {model.GetType().Name}");
+        }
+
+        [Test]
+        public void ThrowsIfUnknownFormat()
+        {
+            ModelReaderWriterOptions options = new ModelReaderWriterOptions("x");
+            Assert.Throws<FormatException>(() => ModelReaderWriter.Write(ModelInstance, options));
+            Assert.Throws<FormatException>(() => ModelReaderWriter.Read<T>(new BinaryData("x"), options));
+
+            Assert.Throws<FormatException>(() => ModelReaderWriter.Write((IPersistableModel<object>)ModelInstance, options));
+            Assert.Throws<FormatException>(() => ModelReaderWriter.Read(new BinaryData("x"), typeof(T), options));
+            if (ModelInstance is IJsonModel<T> jsonModel)
+            {
+                Assert.Throws<FormatException>(() => jsonModel.Write(new Utf8JsonWriter(new MemoryStream()), options));
+                Assert.Throws<FormatException>(() => ((IJsonModel<object>)jsonModel).Write(new Utf8JsonWriter(new MemoryStream()), options));
```

Comment:
Same comment as above - the test should pass without requiring the cast or we have an error.